latex-syntax/src/main/java/org/xwiki/contrib/latex/internal/TemplateProcessor.java

```diff
@@ -159,4 +160,62 @@ public String render(String templateName) throws Exception
     {
         return render(getTemplate(templateName));
     }
+
+    private String getTemplateName(Block block)
+    {
+        // If there's a custom template defined in the Block parameter's, use it!
+        String templateName = block.getParameter("latex-template");
+        if (templateName == null) {
+            templateName = block.getClass().getSimpleName();
+        }
+        return templateName;
+    }
+
+    private Optional<String> renderUIXPs(String templateName, String suffix)
```

Comment:
small note: the UIXP is the extension point. UIXs are implementations of this extension point. So there's one UIXP and several UIXs. So it's either `renderUIXP` or maybe better `renderUIXs` :)